keps/sig-instrumentation/647-apiserver-tracing/README.md

```diff
@@ -168,64 +174,167 @@ GA
-  - How does this feature respond to complete failures of the services on
-    which it depends?  **Traces will stop being exported, and components will store spans in memory until the buffer is full.  After the buffer fills up, spans will be dropped.**
-  - How does this feature respond to degraded performance or high error rates
-    from services on which it depends? **If the bi-directional grpc streaming connection to the collector cannot be established or is broken, the controller retries the connection every 5 minutes (by default).**
-* Monitoring requirements
-  - How can an operator determine if the feature is in use by workloads?  **Operators are generally expected to have access to the trace backend.**
-  - How can an operator determine if the feature is functioning properly?
-  - What are the service level indicators an operator can use to determine the
-    health of the service?  **Error rate of sending traces in the API Server and OpenTelemetry collector.**
-  - What are reasonable service level objectives for the feature?  **Not entirely sure, but I would expect at least 99% of spans to be sent successfully, if not more.**
-* Troubleshooting
-  - What are the known failure modes?  **The API Server is misconfigured, and cannot talk to the collector.  The collector is misconfigured, and can't send traces to the backend.**
-  - How can those be detected via metrics or logs?  Logs from the component or agent based on the failure mode.
-  - What are the mitigations for each of those failure modes?  **None.  You must correctly configure the collector for tracing to work.**
-  - What are the most useful log messages and what logging levels do they require? **All errors are useful, and are logged as errors (no logging levels required). Failure to initialize exporters (in both controller and collector), failures exporting metrics are the most useful.  Errors are logged for each failed attempt to establish a connection to the collector.**
-  - What steps should be taken if SLOs are not being met to determine the
-    problem?  **Look at API Server  and collector logs.**
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: APIServerTracing
+    - Components depending on the feature gate: kube-apiserver
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Does enabling the feature change any default behavior?**
+  No.  The feature is disabled unless a configuration file is specified.
```

Comment:
but even if we switch it on, it doesn't change anything from the "user perspective". We just start sending traces, right?

Might be worth clarifying.

Reply:
To add to this -- the above says it's enabled via a feature flag, rather than a config file. Here, it sounds like we need both. We might want to clarify what happens when flag is on but config file is not specified.

Reply:
Yeah, it doesn't change anything from the user's perspective.  I clarified that.

You do need both.  When the feature gate is enabled, but the config is disabled, the feature is completely disabled.